main.rb

```diff
@@ -0,0 +1,142 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?
+    return (my_all? { |a| !a.nil? }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
+
+  def my_any?
+    return (my_any? { |x| !x.nil? }) unless block_given?
+
+    my_each do |value|
+      return true if yield value
+    end
+    false
+  end
```

Comment:
2. `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. When no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, nil, false].my_any?(1) #should return true
```
...2. When a class is passed as an argument returns true if at least one of the collection is a member of such class.
```ruby
[1, nil, false].my_any?(Integer) #should return true
```
...3. When a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/z/) #should return false
```
...4. When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```ruby
[1, 2 ,3].my_any?(1) #should return true
```